### PR TITLE
Update stdsimd

### PR DESCRIPTION
**Pull request by taiki-e**
r? @gnzlbg

### CODE REVIEW

**Comment by gnzlbg** [comment]
cc @japaric looks like there are some issues while building stdsimd docs. Could you take a look?

**Comment by japaric** [comment]
@gnzlbg it's easy to fix the build failure; what's going to be tricky is showing the right API under `arch::arm` and `arch::aarch64` when building the docs using the "dox" pseudo feature.

The problem is that the *whole* `acle` module is reexported from both `arm` and `aarch64` but the contents of `acle` are architecture specific. The "dox" feature is a trick to render the API docs for all architectures with a single `cargo doc` invocation; if we follow it to the letter for `acle` we'll end up showing ARM specific ACLE API under the `aarch64` module and vice versa.

To show the right API with the "dox" feature we would have to add something like this to `arm/mod.rs` and `aarch64/mod.rs`.

``` rust
// arm/mod.rs

#[cfg(not(dox))] // <- ADD
pub use super::acle::*;

// ADD
#[cfg(dox)]
pub use super::acle::{dmb, dsb, /* plus dozens of other funcions */};
```

Basically re-export the right thing in each module, which is going to be a lot of work to do correctly and maintain.

**Comment by gnzlbg** [comment]
@japaric Would splitting the acle module in, say, `acle_common`, `acle_arm`, and `acle_aarch` fix this ?

**Comment by Centril** [comment]
> @gnzlbg it's easy to fix the build failure; what's going to be tricky is showing the right API under `arch::arm` and `arch::aarch64` when building the docs using the "dox" pseudo feature.

I think the priority for now should be to unblock transitioning libstd to 2018 so that we can finish the transitions.

**Comment by gnzlbg** [comment]
Due to a new major feature landing on stdsimd this week (unfortunate timing), the library does not build in tree anymore. 

What @taiki-e is doing is IMO the best way to proceed. 

Let's get stdsimd in Rust2015 mode building in tree again, which is not trivial (@taiki-e do you mind if I take over doing this?).  

Once that is merged, the Rust2018 PR for libstd can be updated, and we can just focus on solving Rust2018 related issues there. @taiki-e has already sent a PR fixing some of this, so I'm sure they will be able to nail all other issues in no time once we are there.

**Comment by taiki-e** [comment]
@gnzlbg 

> (@taiki-e do you mind if I take over doing this?)

No, go ahead.

**Comment by gnzlbg** [comment]
@taiki-e i've sent a PR for the update #58675 . I'm not sure if those changes are enough for it to work, but I'll iterate back and forth until that gets merged.

**Comment by taiki-e** [comment]
@gnzlbg Thanks!
